x-pack/plugins/ml/common/util/job_utils.ts

```diff
@@ -182,16 +199,16 @@ export function isModelPlotEnabled(job, detectorIndex, entityFields) {
 
 // Returns whether the version of the job (the version number of the elastic stack that the job was
 // created with) is greater than or equal to the supplied version (e.g. '6.1.0').
-export function isJobVersionGte(job, version) {
-  const jobVersion = _.get(job, 'job_version', '0.0.0');
+export function isJobVersionGte(job: CombinedJob, version: string): boolean {
+  const jobVersion = job.job_version ?? '0.0.0';
   return semver.gte(jobVersion, version);
 }
 
 // Takes an ML detector 'function' and returns the corresponding ES aggregation name
 // for querying metric data. Returns null if there is no suitable ES aggregation.
 // Note that the 'function' field in a record contains what the user entered e.g. 'high_count',
 // whereas the 'function_description' field holds an ML-built display hint for function e.g. 'count'.
-export function mlFunctionToESAggregation(functionName) {
+export function mlFunctionToESAggregation(functionName: string): string | null {
```

Comment:
As you are already doing refactoring of this file, could you also please switch all the hard-coded function names to use the enums from `kibana/x-pack/plugins/ml/common/constants/aggregation_types.ts`. They are used in this function, plus some in `isModelPlotChartableForDetector`. The ml functions should be switched to use `ML_JOB_AGGREGATION` and the ES return values should come from `ES_AGGREGATION`.

Reply:
Check https://github.com/elastic/kibana/pull/66265/commits/692f7a6955915da8b3d0644e1e7b5326177ad64e